### PR TITLE
feat(otkit-icons): adds themeable icons support

### PR DESCRIPTION
**Pull request by ronniechong**
This PR is to add a set of otkit-icons that supports themeable SVG icons. The SVG icon colors will have `currentColor` for values (`fill` and `stroke`) which allows the icon to inherit colors from its parent element.

There will be no breaking change in the current `otkit-icons` token. I simply added additional importable files `token.theme.common.js` / `token.theme.scss` / `token.theme.cssmodules.css` into the package.

In essence, there'll be two tokens for the same icons:
```javascript
// Imports standard icons
import { icTicket } from 'otkit-icons/token.common'

// Imports themeable icons
import { icTicket } from 'otkit-icons/token.theme.common'
```

Limitations:
- Some SVG may have specific colors (e.g Google, Facebook, Twitter icons) and the `token.theme.common` will strip those colors and replaced with `currentColor`

Notes:
- The Issue was discussed in https://github.com/opentable/design-tokens/issues/452
- SCSS, CSS Modules and CommonJS are supported.
- README is documented on its usage
- Style guide for themeable icons is not implemented yet. Possibly we need to discuss with the design team if we need to update the `otkit-icons` page to accommodate two different set of icons.

**To discusss:**
- Bumping this version as a major
- How do we achieve this
- Inform everyone (PSA) of the change

### CODE REVIEW

File: utils/build-icons/index.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
+const Promise = require('bluebird');
+
 const generateTokenFile = require('./lib/generateTokenFile');
+const generateThemeableTokenFile = require('./lib/generateThemeableTokenFile');
```
Comment (ronniechong) [inline utils/build-icons/index.js]: Making these two imports separately so in the future we can easily take one out if we no longer needs it.

File: package.json [hunk 2]
```diff
@@ -28,21 +28,26 @@
...
       "delay": 0
     }
   },
   "scripts": {
     "build": "npm run build-ottheme && npm run build-otkit",
     "build-icons": "lerna exec --scope *-icons -- node '$LERNA_ROOT_PATH/utils/build-icons'",
-    "build-otkit": "npm run build-icons && lerna exec --scope otkit-* theo token.yml -- --format scss,cssmodules.css,common.js --dest .",
+    "build-otkit": "npm run build-icons && npm run build-otkit-all && npm run build-otkit-theme-icon",
+    "build-otkit-all": "lerna exec --scope otkit-* theo token.yml -- --format scss,cssmodules.css,common.js --dest .",
+    "build-otkit-theme-icon": "lerna exec --scope otkit-icons theo token.theme.yml -- --format scss,cssmodules.css,common.js --dest .",
```
Comment (ronniechong) [inline package.json]: Adding extra script to run `theo` solely for the themeable icons

**Comment by jakeclements** [comment]
Hey @ronniechong, can we merge this?

**Comment by ronniechong** [comment]
Had a conversation with Vincent. We are working on a strategy to force a major version bump starting with this  https://github.com/opentable/design-tokens/pull/463

**Pull request merged**